packages/atomic/src/components/search/result-template-components/atomic-result-localized-text/atomic-result-localized-text.ts

```diff
@@ -0,0 +1,81 @@
+import {isNullOrUndefined} from '@coveo/bueno';
+import {Result, ResultTemplatesHelpers} from '@coveo/headless';
+import {Component, Prop, State} from '@stencil/core';
+import {
+  InitializableComponent,
+  InitializeBindings,
+} from '../../../../utils/initialization-utils';
+import {MapProp} from '../../../../utils/props-utils';
+import {Bindings} from '../../atomic-search-interface/atomic-search-interface';
+import {ResultContext} from '../result-template-decorators';
+
+/**
+ * The `atomic-result-localized-text` component renders a target i18n localized string using field values as parameter.
+ *
+ * Given this i18n configuration:
+ * ```
+ * searchInterface.i18n.addResourceBundle('en', 'translation', {
+ *    foo: 'Hello world {{a_field_value}} !',
```

Comment:
Should `a_field_value` be `a_field_name`?
So, for instance, if you want the name of an author to show (e.g., J.K Rowling), then you'd write:
`'Hello world {{author}} !'`, not `'Hello world {{J.K. Rowling}} !'`.
In other words, what users want to specify here is the name of the field (e.g., `author`) rather than a value of the field (`J.K. Rowling`).

Reply:
The example is how it's supposed to work, but probably badly phrased if it's confiusing for you.

The way it works is that you need to define two things:

* What is the placeholder/variable that should be replaced in each i18n string. Here that would be `{{a_field_value}}`. It could have been named anything.

* What is the field that should be used to replace that variable/placeholder. This cannot be named arbitrarily: it has to be a valid field on a result.

So when you configure it like so:

```
<atomic-result-localized-text field-value-author="a_field_value" ></atomic-result-localized-text>
```

It means that the value of the field `author` will be used to replace the placeholder `{{a_field_value}}`. With that explanation, how would you rephrase the documentation/example above.